### ISSUE TITLE
Rate limit GC speed with rate_limter

### ISSUE THREAD

**Issue by yiwu-arbug** [tikv/titan#50]
Limit GC and Titan read IO with rate_limiter, to avoid burst of IO starve online read and write.

**Comment by hykych** [comment]
@JiayuZzz Hi, May I ask why did you close this issue without comment? we came across this issue also.

**Comment by yiwu-arbug** [comment]
Hi, @hykych . sorry for not linking the corresponding PR, but after #54 Titan respect `db_opts.rate_limier` with mode = kReadsOnly or kAllIo. Thanks.

**Comment by hykych** [comment]
@yiwu-arbug Thanks for the quick response. I tried the config in Tidb [doc](https://docs.pingcap.com/tidb/stable/titan-configuration) about Titan, ```rate-bytes-per-sec = 30MB```. But seems like Titan doesn't respect this config. Do I get it wrong?
the TiDB version is 4.0.4

**Comment by hykych** [comment]
> @yiwu-arbug Thanks for the quick response. I tried the config in Tidb [doc](https://docs.pingcap.com/tidb/stable/titan-configuration) about Titan, `rate-bytes-per-sec = 30MB`. But seems like Titan doesn't respect this config. Do I get it wrong?
> the TiDB version is 4.0.4

@yiwu-arbug  I got the reason now. it's because the default ```rate-limiter-mode``` is ```WriteOnly```. After setting the value to ```AllIo```, the config works.

**Comment by yiwu-arbug** [comment]
@hykych there are documentation https://docs.pingcap.com/tidb/stable/tikv-configuration-file#rate-limiter-mode

**Comment by hykych** [comment]
> https://docs.pingcap.com/tidb/stable/tikv-configuration-file#rate-limiter-mode

@yiwu-arbug True, but doesn't mention Titan GC's rate-limit only works when ```rate-limiter-mode``` is ```ReadOnly``` or ```AllIo```

**Comment by yiwu-arbug** [comment]
@hykych actually the rate limiter should also work for `kWritesOnly` mode. maybe in your case there are a lot of data deletion so that the read IO for GC is way larger than write IO?

**Comment by hykych** [comment]
> @hykych actually the rate limiter should also work for `kWritesOnly` mode. maybe in your case there are a lot of data deletion so that the read IO for GC is way larger than write IO?

the read IO for GC is larger than write IO, but write IO is also larger than the ```rate-bytes-per-sec``` , I observed the panel```Blob GC bytes flow```

**Comment by yiwu-arbug** [comment]
@hykych could be the rate limiter not working or the Blob GC bytes flow inacurate.
